From 2e84550b47b475d830622f41576992a1d45d6af0 Mon Sep 17 00:00:00 2001
From: Zhao Yi <38274519+yizhao1@users.noreply.github.com>
Date: Tue, 15 Mar 2022 19:28:26 +0800
Subject: [PATCH] Makefile: fix parallel build failure (#267)

* Makefile: fix parallel build failure

Add src/_features.h as dependency for dbus.o to fix the parallel build
failure:
src/dbus.c:17:10: fatal error: _features.h: No such file or directory

Upstream-Status: Backport
[https://github.com/canonical/netplan/commit/2e84550b47b475d830622f41576992a1d45d6af0]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 0368e41..8ac7432 100644
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,7 @@ NOSETESTS3 ?= $(shell command -v nosetests-3 || command -v nosetests3 || echo tr
 
 default: netplan/_features.py generate netplan-dbus dbus/io.netplan.Netplan.service doc/netplan.html doc/netplan.5 doc/netplan-generate.8 doc/netplan-apply.8 doc/netplan-try.8 doc/netplan-dbus.8 doc/netplan-get.8 doc/netplan-set.8
 
-%.o: src/%.c
+%.o: src/%.c src/_features.h
 	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -c $^ `pkg-config --cflags --libs glib-2.0 gio-2.0 yaml-0.1 uuid`
 
 libnetplan.so.$(NETPLAN_SOVER): $(SRCS) abicompat.lds
@@ -62,7 +62,7 @@ libnetplan.so.$(NETPLAN_SOVER): $(SRCS) abicompat.lds
 generate: libnetplan.so.$(NETPLAN_SOVER) generate.o
 	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(filter-out $<,$^) -L. -lnetplan `pkg-config --cflags --libs glib-2.0 gio-2.0 yaml-0.1 uuid`
 
-netplan-dbus: libnetplan.so.$(NETPLAN_SOVER) src/_features.h dbus.o
+netplan-dbus: libnetplan.so.$(NETPLAN_SOVER) dbus.o
 	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(filter-out $<,$(patsubst %.h,,$^)) -L. -lnetplan `pkg-config --cflags --libs libsystemd glib-2.0 gio-2.0 yaml-0.1 uuid`
 
 src/_features.h: src/[^_]*.[hc]
-- 
2.25.1

